Support EMACS_SERVER_SOCKET_DIR in emacsclient.
Patch by Martin Vaeth <martin at mvath.de> from Sep 7, 2014; update Sep 12, 2020
--- 1/lib-src/emacsclient.c
+++ 1/lib-src/emacsclient.c
@@ -500,6 +500,16 @@
 {
   alternate_editor = egetenv ("ALTERNATE_EDITOR");
   tramp_prefix = egetenv ("EMACSCLIENT_TRAMP");
+#ifdef SOCKETS_IN_FILE_SYSTEM
+  const char *sockdir = egetenv("EMACS_SERVER_SOCKET_DIR");
+  if(sockdir)
+    {
+       char *buf = (char *) xmalloc (strlen (sockdir) + 7 + 1);
+       strcpy (buf, sockdir);
+       strcat (buf, "/server");
+       socket_name = buf;
+    }
+#endif
 
   while (true)
     {
